pkg/eval/errs/errs.go

```diff
@@ -12,15 +12,20 @@ type OutOfRange struct {
 	ValidLow  int
 	ValidHigh int
 	Actual    string
+	ValidFmt  string
```

Comment:
This is fine from the perspective of code building `OutOfRange`, but it introduces information that is not useful for consumers of this data type; I'd change `ValidLow` and `ValidHigh` to strings.

Reply:
> I'd change ValidLow and ValidHigh to strings.

Sure, that works too, but my approach avoids having to add two `fmt.Sprintf` calls at every site which constructs this object. And since the vast majority will continue to use the default decimal representation they don't even need to pass a format string to the constructor.